distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/APMJvmOptions.java

```diff
@@ -208,12 +209,35 @@ private static Map<String, String> extractDynamicSettings(Map<String, String> pr
         return cliOptionsMap;
     }
 
-    private static Map<String, String> extractApmSettings(Settings settings) throws UserException {
+    // package private for testing
+    static Map<String, String> extractApmSettings(Settings settings) throws UserException {
         final Map<String, String> propertiesMap = new HashMap<>();
 
         final Settings agentSettings = settings.getByPrefix("tracing.apm.agent.");
         agentSettings.keySet().forEach(key -> propertiesMap.put(key, String.valueOf(agentSettings.get(key))));
 
+        // special handling of global labels, the agent expects them in format: key1=value1,key2=value2
+        final Settings globalLabelsSettings = settings.getByPrefix("tracing.apm.agent.global_labels.");
+        final StringBuilder globalLabels = new StringBuilder();
+
+        for (var globalLabel : globalLabelsSettings.keySet()) {
+            // remove the individual label from the properties map, they are harmless, but we shouldn't be passing
+            // something to the agent it doesn't understand.
+            propertiesMap.remove("global_labels." + globalLabel);
+            var globalLabelValue = globalLabelsSettings.get(globalLabel);
+            if (Strings.isNotBlank(globalLabelValue)) {
+                // sanitize for the agent labels separators in case the global labels passed in have , or =
+                globalLabelValue = globalLabelValue.replaceAll("[,=]", "_");
+                // append to the global labels string
+                globalLabels.append(globalLabel).append('=').append(globalLabelValue).append(",");
+            }
+        }
+
+        if (globalLabels.length() > 0) {
+            globalLabels.setLength(globalLabels.length() - 1); // remove trailing comma
```

Comment:
Argh!!! trailing comma - easy to miss. Another option is:

```
  final StringJoiner globalLabels = new StringJoiner(",");

  if (Strings.isNullOrBlank(globalLabelValue) == false) {
      ...
      // append to the global labels string
      globalLabels.add(String.join("=", globalLabel, globalLabelValue));
 }
```

Then there is guaranteed to be no trailing comma :-)

P.S. while largely benign, we prob wanna asset this, e.g. 
  `hasEntry(equalTo("global_labels"), not(endsWith(","))), // ensure no trailing comma`